### PR TITLE
feat: don't panic unbolding feature

### PR DESCRIPTION
**Pull request by John-Getachew**
## Description
- encapsulated the unbolding function so it could be tested alone and so it could be reusable in the bottom icon row.
- Followed project conventions by adding enums to LabelText.ts and TestID.ts (rather than hard-coding text).
- Used Js functions to illustrate functionality rather than loops or recursions (since this challenge is for a JS react role :) ).
- Didn't add comment and doc since the change is fairly easy to read and code is usually good when it can be understood by itself.

### Testing

- [ ] Cypress click test has been added to the newly added UI.
- [ ] Jest test has been added to test helper function functionality.

### Suggestions
- Dev team needs to maintain the app (regarding dependencies and node versions) so that the web app is compatible withe new node versions. (E.g. Node v16)

### CODE REVIEW

File: tests/unit/client/utils/index.test.ts [hunk 1]
```diff
@@ -107,4 +112,16 @@ describe('Utilities', () => {
       expect(getActiveNoteFromShortUuid(notes, shortActiveNoteId)).toEqual(undefined)
     })
   })
+  describe('removeBoldStringsFromNote', () => {
+    test(`should remove astrix from both sides of a word`, () => {
```
Comment (mknecht) [inline tests/unit/client/utils/index.test.ts:116]:
```suggestion
    test(`should remove asterisk from both sides of a word`, () => {
```

File: src/client/utils/helpers.ts [hunk 2]
```diff
@@ -294,3 +294,10 @@ export const getNoteBarConf = (
 export const copyToClipboard = (text: string) => {
   clipboard.writeText(text)
 }
+
+export const unboldNote = (note: string) => {
+  var unboldedText = note.replace(/\*\*([^*]*(?:\*(?!\*)[^*]*)*)\*\*/g, function (match, index) {
```
Comment (mknecht) [inline src/client/utils/helpers.ts:299]: fairly sure this regex also cleans up stuff **that is not ** bold :) https://www.markdownguide.org/basic-syntax/

File: tests/unit/client/utils/index.test.ts [hunk 3]
```diff
@@ -107,4 +112,16 @@ describe('Utilities', () => {
...
+      const boldNote = `This is your **world**. This is gonna be a **happy** little seascape.`
+      const unboldedNote = `This is your world. This is gonna be a happy little seascape.`
+      expect(unboldNote(boldNote)).toEqual(unboldedNote)
+    })
+    test(`should not remove two astrixes that aren't followed by a word and ending two astrixes`, () => {
```
Comment (mknecht) [inline tests/unit/client/utils/index.test.ts:121]: yup! in reality a few more edge cases might be covered. :)

File: tests/e2e/utils/testNotesHelperUtils.ts [hunk 4]
```diff
@@ -121,6 +121,10 @@ const clickNoteOptionCopyLinkedNoteMarkdown = () => {
   clickTestID(TestID.COPY_REFERENCE_TO_NOTE)
 }
 
+const clickNoteOptionUnboldNote = () => {
```
Comment (mknecht) [inline tests/e2e/utils/testNotesHelperUtils.ts:124]: That's in the description:

>  Cypress click test has been added to the newly added UI.

and you prepared for it by creating this click helper (in line with the code base). But where do you use it? See other tests in tests/e2e/integration/note.test.ts
